packages/main/src/plugin/authentication.ts

```diff
@@ -0,0 +1,189 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import type {
+  authentication,
+  AuthenticationProvider,
+  AuthenticationSession,
+  AuthenticationSessionsChangeEvent,
+  AuthenticationGetSessionOptions,
+  Event,
+  AuthenticationProviderAuthenticationSessionsChangeEvent,
+  AuthenticationSessionAccountInformation,
+  AuthenticationProviderOptions,
+  Disposable,
+} from '@podman-desktop/api';
+// import { window } from '@podman-desktop/api';
+import { Emitter } from './events/emitter';
+type Authentication = typeof authentication;
+
+/**
+ * Structure to save authentication provider information
+ * with additional metadata
+ */
+export interface ProviderWithMetadata {
+  id: string;
+  label: string;
+  provider: AuthenticationProvider;
+  options: AuthenticationProviderOptions;
+}
+
+export interface AuthenticationProviderInfo {
+  id: string;
+  displayName: string;
+  accounts: AuthenticationSessionAccountInformation[];
+}
+
+export interface ExtensionInfo {
+  id: string;
+  label: string;
+}
+
+export interface AllowedExtension {
+  id: string;
+  name: string;
+  allowed?: boolean;
+}
+
+export class AuthenticationImpl {
+  private _authenticationProviders: Map<string, ProviderWithMetadata> = new Map<string, ProviderWithMetadata>();
+
+  constructor(private apiSender: any) {}
```

Comment:
since yesterday we have a type
https://github.com/containers/podman-desktop/pull/1888